code-challenges.test.js

```diff
@@ -52,18 +128,80 @@ const randomNouns2 = ["temperature", "database", "chopsticks", "mango", "deducti
 
 // b) Create the function that makes the test pass.
 
-
-
+// Pseudo Code
+// Create a function and name it firstCaps
+// Set a parameter called array
+// Write the code that will take an array of words as an input and return an an array with all words capitalized
+// Return a mapped array using .map()
+// Set the map value to array and identify the first character of each word using charAt(0)
+// Capitalize first letter using toUpperCase()
+// Join capitalized first letter with rest of string using + .slice(1)
+// Call the function twice, passing in randomNouns1 and randomNouns2 as arguments
+const firstCaps = (array) => {
+  return array.map(array => array.charAt(0).toUpperCase() + array.slice(1))
+}
+
+firstCaps(randomNouns1)
+firstCaps(randomNouns2)
 // --------------------3) Create a function that takes in a string and logs the index of the first vowel.
 
 // a) Create a test with expect statements for each of the variables provided.
 
+describe("firstVowel", () => {
+  it("returns first value in a string based on input", () => {
+    expect(firstVowel(vowelTester1)).toEqual(1)
+    expect(firstVowel(vowelTester2)).toEqual(0)
+    expect(firstVowel(vowelTester3)).toEqual(2)
+  })
+})
+
+// FAIL  ./code-challenges.test.js
+//  firstVowel
+//    ✕ returns first value in a string based on input
+//
+//  ● firstVowel › returns first value in a string based on input
+//
+//    ReferenceError: firstVowel is not defined
+//
+//      149 | describe("firstVowel", () => {
+//      150 |   it("returns first value in a string based on input", () => {
+//    > 151 |     expect(firstVowel(vowelTester1)).toEqual(1)
+//          |     ^
+//      152 |     expect(firstVowel(vowelTester2)).toEqual(0)
+//      153 |     expect(firstVowel(vowelTester3)).toEqual(2)
+//      154 |   })
+//
+//      at Object.<anonymous> (code-challenges.test.js:151:5)
+//
+// Test Suites: 1 failed, 1 total
+// Tests:       1 failed, 1 total
+// Snapshots:   0 total
+// Time:        0.145 s, estimated 1 s
+
 const vowelTester1 = "learn"
 // Expected output: 1
 const vowelTester2 = "academy"
 // Expected output: 0
 const vowelTester3 = "challenges"
 // Expected output: 2
 
+// Pseudo Code
+// Create a function called firstVowel
+// Set a parameter called string
+// Write the code that will take in a string and log the index of the first vowel
+// Write a for loop, setting the length to i < string.length
+// Write a conditional statement which says if the index of the string is equal to a value, return that index
+// Call the function three times, passing in vowelTester1, vowelTester2, and vowelTester3 as arguments
 
 // b) Create the function that makes the test pass.
+const firstVowel = (string) => {
+  for(let i = 0; i < string.length; i++){
+    if (string[i] == "a" || string[i] == "e" || string[i] == "i" || string[i] == "o" || string[i] == "u") {
+    return string.indexOf(string[i])
+    }
+  }
+}
```

Comment:
You got it!  Generally, I would recommend using strict equality when using conditionals.